embedded/store/immustore.go

```diff
@@ -1693,29 +1693,18 @@ func (s *ImmuStore) ReadTx(txID uint64, tx *Tx) error {
 
 // ReadValue returns the actual associated value to a key at a specific transaction
 // ErrExpiredEntry is be returned if the specified time has already elapsed
-// Note; the tx is read to be safe in cases provided tx was manipulated at runtime
-func (s *ImmuStore) ReadValue(txID uint64, txHolder *Tx, key []byte) ([]byte, error) {
-	if txHolder == nil {
+func (s *ImmuStore) ReadValue(entry *TxEntry) ([]byte, error) {
+	if entry == nil {
```

Comment:
We should check if it is read only, right?

Reply:
yes, also ensuring readonly metadata cannot be externally instantiated